### PR TITLE
Add files via upload

### PR DESCRIPTION
**Pull request by manokhina**

### CODE REVIEW

File: binarysearch.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
+# implementation of binary search algorithm
+
+# binary search > naive search
+
+# naive search: scan entire list and ask if its equal to the target
+# if yes, return the index
+# if no, return -1
+
+def naive_search(l, target):
```
Comment (manokhina) [inline binarysearch.py:9]: It is not advised to use letters like `l` as variable names as they might be confused with numbers

File: binarysearch.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    for i in range(len(l)):
+        if l[i] == target:
+            return i
+    return -1
+
+
+# binary search uses divide and conquer!
+# we will leverage the fact that our list is SORTED
+def binary_search(l, target, low=None, high=None):
+    if low is None:
```
Comment (manokhina) [inline binarysearch.py:20]: Can be written shorter: `if not low`

File: binarysearch.py (hunk 1 above)
Comment (jazbassi1) [reply binarysearch.py:9]: On pycharm it shows the l in a more distinctive way where it does not look like number but I see on github it shows it looks like the number 1 for example in this case. But will bear this in mind

File: binarysearch.py (hunk 2 above)
Comment (jazbassi1) [reply binarysearch.py:20]: This makes sense. Will note this

**Pull request merged**